<?xml version='1.0'?>
<launch>

  <!-- load the controllers -->

    <rosparam file="$(find hunter2_control)/config/hunter2_controller.yaml" command="load" />
  <node name="controller_manager" pkg="controller_manager" type="spawner" respawn="false"
        output="screen" ns="/hunter2" args="joint_state_controller 
                                                                                      wheel_joint1_velocity_controller wheel_joint2_velocity_controller
                                                                                      wheel_joint3_velocity_controller wheel_joint4_velocity_controller
                                                                                      wheel_joint3_steering_position_controller wheel_joint4_steering_position_controller"/>

  <!-- convert joint states to TF transforms for rviz, etc -->

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="false" output="screen">
    <remap from="/joint_states" to="/hunter2/joint_states" />
      </node>
    <!-- 运行robot_state_publisher节点，发布tf  -->
    <!--<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"  output="screen" >
        <param name="publish_frequency" type="double" value="50.0" />
    </node>-->

  <node pkg="hunter2_control" type="servo_commands.py" name="servo_commands" output="screen"/>
  <node pkg="hunter2_gazebo" name="gazebo_odometry_node" type="gazebo_odometry.py"/>


</launch>
